client/allocrunner/taskrunner/logmon_hook_test.go

```diff
@@ -55,10 +55,10 @@ func TestTaskRunner_LogmonHook_LoadReattach(t *testing.T) {
 	require.Equal(t, orig, cfg)
 }
 
-// TestTaskRunner_LogmonHook_StartStop asserts that a new logmon is created the
-// first time Prestart is called, reattached to on subsequent restarts, and
-// killed on Stop.
-func TestTaskRunner_LogmonHook_StartStop(t *testing.T) {
+// TestTaskRunner_LogmonHook_StartRestartStop asserts that a new logmon is
+// created the first time Prestart is called, reattached to on subsequent
```

Comment:
```go
// a new logmon is created the first time Prestart is called
```

To me `first time` implies that a single logmon process should be reused and reconnected to between restarts. Mind rewriting this comment to make sure it's unambiguous?

> TestTaskRunner_LogmonHook_StartRestartStop asserts that logmon's lifecycle matches the tasks: it is killed when the task exits and started (or restarted or reattached) whenever the task starts.

Read on for context if your bored :sweat_smile:

<details>
I think logmon was written such that the process lived for the lifetime of the *alloc,* not the lifetime of an individual task invocation.

Most hooks that are implemented that way do so by setting `response.Done = true` somewhere in Prestart so that TaskRunner never bothers calling it again. However logmon does not since it *does* need to *reconnect* to the running logmon process on task restart or restart logmon if the entire host was restarted and logmon no longer exists...

...and because we have to handle that `restart logmon if it did die` case, I think just killing it every time will work fine functionally!
</details>